teos/src/config.rs

```diff
@@ -164,6 +177,24 @@ pub struct Config {
 }
 
 impl Config {
+    pub fn get_auth_method(&self) -> AuthMethod {
+        if self.btc_rpc_user != String::new() && self.btc_rpc_password != String::new() {
+            if self.btc_rpc_cookie != String::new() {
+                AuthMethod::Multiple
+            } else {
+                AuthMethod::UserPass
+            }
+        } else if self.btc_rpc_cookie != String::new() {
+            if self.btc_rpc_user != String::new() || self.btc_rpc_password != String::new() {
+                AuthMethod::Multiple
+            } else {
+                AuthMethod::CookieFile
+            }
+        } else {
+            AuthMethod::Invalid
+        }
+    }
```

Comment:
```suggestion
    pub fn get_auth_method(&self) -> AuthMethod {
        let (user, pass, cookie) = (
            !self.btc_rpc_cookie.is_empty(),
            !self.btc_rpc_user.is_empty(),
            !self.btc_rpc_password.is_empty(),
        );
        if user && pass && cookie {
            AuthMethod::Multiple
        } else if cookie {
            AuthMethod::CookieFile
        } else if user && pass {
            AuthMethod::UserPass
        } else {
            AuthMethod::Invalid
        }
    }
```

Reply:
or yet better:
```rust
    pub fn get_auth_method(&self) -> AuthMethod {
        let (user, pass, cookie) = (
            !self.btc_rpc_cookie.is_empty(),
            !self.btc_rpc_user.is_empty(),
            !self.btc_rpc_password.is_empty(),
        );
        match (user, pass, cookie) {
            (true, true, true) => AuthMethod::Multiple,
            (true, true, false) => AuthMethod::UserPass,
            (false, false, true) => AuthMethod::CookieFile,
            (_, _, _) => AuthMethod::Invalid,
        }
    }
```

which makes sure we don't supply only `user`-only or `pass`-only when the `CookieFile` is active.

Reply:
I just noticed you wanna brand the user only + cookie or pass only + cookie as multiple (which makes more sense given the printed error messages), so this should be:

```rust
    pub fn get_auth_method(&self) -> AuthMethod {

        let user = !self.btc_rpc_user.is_empty();
        let pass = !self.btc_rpc_password.is_empty();
        let cookie = !self.btc_rpc_cookie.is_empty();

        match (user, pass, cookie) {
            (false, false, false) => AuthMethod::Invalid,
            (true, true, false) => AuthMethod::UserPass,
            (false, false, true) => AuthMethod::CookieFile,
            (_, _, _) => AuthMethod::Multiple,
        }
    }
```